ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -516,31 +865,84 @@ public Container getContainerByName(String accountName, String containerName) th
       return null;
     }
 
-    Account account = getAccountByName(accountName);
-    if (account == null) {
-      return null;
-    }
-    Container container = account.getContainerByName(containerName);
-    if (container == null) {
-      // If container is not present in the cache, query from mysql db
-      try {
-        container = mySqlAccountStore.getContainerByName(account.getId(), containerName);
-        if (container != null) {
-          // write container to in-memory cache
-          updateContainersInCache(Collections.singletonList(container));
-          logger.info("Container {} in Account {} is not found locally; Fetched from mysql db", containerName,
-              accountName);
-          accountServiceMetrics.onDemandContainerFetchCount.inc();
-        } else {
-          // Add account_container to not-found LRU cache
-          recentNotFoundContainersCache.add(accountName + SEPARATOR + containerName);
-          logger.error("Container {} is not found in Account {}", containerName, accountName);
+    if (config.enableNewDatabaseForMigration) {
+      Account account = getAccountByNameNew(accountName);
+      Container container = null;
+      if (account != null) {
+        container = account.getContainerByName(containerName);
+      }
+      if (container == null) {
+        // If container is not present in the cache, query from mysql db
+        try {
+          container = mySqlAccountStoreNew.getContainerByName(account.getId(), containerName);
+          if (container != null) {
+            // write container to in-memory cache
+            updateContainersInCacheNew(Collections.singletonList(container));
+            logger.info("Container {} in Account {} is not found locally; Fetched from mysql db", containerName,
+                accountName);
+            accountServiceMetrics.onDemandContainerFetchCountNew.inc();
+          }
+        } catch (SQLException e) {
+          throw translateSQLException(e);
+        }
+        logger.trace("Found container from new db, container: " + containerName);
+        return container;
```

Comment:
This block is repeated many times for old and new account stores. Can you move it to a new helper function and call it with different parameters?

Reply:
Add the helper function for old db and later will remove it after migration is fully tested.